hnix-store-tests/tests/SignatureSpec.hs

```diff
@@ -0,0 +1,18 @@
+module SignatureSpec where
+
+import Test.Hspec (Spec, describe, shouldBe)
+import Test.Hspec.QuickCheck (prop)
+import Test.QuickCheck (Arbitrary(arbitrary), forAll)
+
+import System.Nix.Arbitrary ()
+
+import System.Nix.Signature
+
+spec :: Spec
+spec = do
+  describe "Signature" $ do
+    prop "roundtrips" $
+      forAll arbitrary $ \p -> do
+        let encoded = signatureToText p
+            parsed = parseSignature encoded
```

Comment:
`forAll arbitrary` is not needed here and you can simply do `prop "roundtrips" $ \p ->`. `prop` is basically a shortcut for `it "..." $ property $ ..`  since it is quite common one.

When you write QC props you can write
```haskell
myProp :: Int -> Bool -> Property
myProp xi yb = ...
```
and when you i.e. need to restrict `Int`s using a modifier like `NonZero` or `suchThat` you switch it to `forAll`

```haskell
myProp :: Bool -> Property
myProp yb = forAll (arbitrary `suchThat` (/=42)) $ ...
```

Now since roundtrip is a common pattern as well, you can use recently added [`Test.Hspec.Nix.roundtrips`](https://github.com/haskell-nix/hnix-store/blob/master/hnix-store-tests/src/Test/Hspec/Nix.hs) combinator.

Reply:
Thanks - updated to use `Test.Hspec.Nix.roundtrips`.